persist/fs/msgpack/decoder.go

```diff
@@ -360,47 +366,55 @@ func (dec *Decoder) decodeLogMetadata() schema.LogMetadata {
 	return logMetadata
 }
 
-func (dec *Decoder) decodeRootObject(expectedVersion int, expectedType objectType) int {
-	dec.checkVersion(expectedVersion)
+func (dec *Decoder) decodeRootObject(expectedVersion int, expectedType objectType) (int, int) {
```

Comment:
Maybe a good idea to name the result ints in the signature to be clear what’s being returned.  Either that or change it to a struct result type which will have named fields so it’s clear what’s being returned.

Reply:
yeah sorry I hacked this together when I was testing something on the test cluster and didn't come back and clean it up properly. I'll use named returns in the signature

Reply:
done